### PR TITLE
feat: add molecule parser tools

### PR DESCRIPTION
**Pull request by PainterPuppets**
resolve: https://github.com/Magickbase/ckb-explorer-public-issues/issues/660

image

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -5,9 +5,10 @@
   "dependencies": {
     "@ant-design/icons": "4.8.1",
     "@ckb-lumos/base": "0.22.2",
+    "@ckb-lumos/codec": "0.22.2",
     "@ckb-lumos/config-manager": "0.22.2",
     "@ckb-lumos/helpers": "0.22.2",
-    "@ckb-lumos/lumos": "^0.22.2",
```
Comment (Keith-CY) [inline package.json:10]: Why remove `@ckb-lumos/lumos`? I thought most functions can be imported from the top package `@ckb-lumos/lumos`

File: package.json [hunk 2]
```diff
@@ -42,13 +43,17 @@
...
     "react-resize-detector": "7.1.2",
     "react-router": "5.3.4",
     "react-router-dom": "5.3.4",
     "react-scripts": "5.0.1",
     "sass": "1.77.1",
-    "styled-components": "5.3.11"
+    "string-width": "4.2.3",
+    "strip-ansi": "6.0.1",
+    "styled-components": "5.3.11",
+    "wrap-ansi": "7.0.0"
```
Comment (Keith-CY) [inline package.json:56]: These 3 new dependents are not used in the codebase

File: src/components/ui/Alert.tsx [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+//       variant: {
+//         default: 'bg-background text-foreground',
+//         destructive: 'border-destructive/50 text-destructive dark:border-destructive [&>svg]:text-destructive',
+//       },
+//     },
+//     defaultVariants: {
+//       variant: 'default',
+//     },
+//   },
+// )
```
Comment (Keith-CY) [inline src/components/ui/Alert.tsx:19]: This comment can be removed if it's not for demonstration.

**Comment by Keith-CY** [comment]
IMO, mentioning `built-in schemas ` next to the title `Input schema(mol)` makes it outstanding.

**Comment by Keith-CY** [comment]
The `built-in` schema could be parsed and the first item could be active by default

image

File: package.json (hunk 1 above)
Comment (PainterPuppets) [reply package.json:10]: I thought `@ckb-lumos/lumos` is a collection of other `lumos` packages that will bring in all the sub-packages, some of which are brought in with some version dependency issues. functions currently imported from `@ckb-lumos/lumos` can also be import in directly from sub-packages, so i remove this top-level package

File: package.json (hunk 2 above)
Comment (homura) [reply package.json:56]: [Here](https://github.com/ckb-js/lumos/blob/c41f602d1ce47c8ee656c871ed960979619c694b/examples/molecule-codegen/package.json#L7) is an example of `lumos/molecule` without explicitly installing these packages, and it works. Could you please share the error message when dropping these dependencies?

Comment (PainterPuppets) [reply package.json]: > [Here](https://github.com/ckb-js/lumos/blob/c41f602d1ce47c8ee656c871ed960979619c694b/examples/molecule-codegen/package.json#L7) is an example of `lumos/molecule` without explicitly installing these packages, and it works. Could you please share the error message when dropping these dependencies?

I deleted it and tried to build it and found it was fine... I'm sorry for wasting your time.

**Comment by Keith-CY** [comment]
> The `built-in` schema could be parsed and the first item could be active by default
> 
> image

Any update on this suggestion?

**Comment by Keith-CY** [comment]
The suggestion above will be implemented in another PR

**Review by Keith-CY — APPROVED** [review]

**Comment by PainterPuppets** [comment]
> > The `built-in` schema could be parsed and the first item could be active by default
> > image
> 
> Any update on this suggestion?

Just saw that, it be implemented at these lines

https://github.com/Magickbase/ckb-explorer-frontend/pull/339/files#diff-51e3a32fe74b2924baa18f7966fb3867256d844a53c58339ffb037301f220b28R17-R20

**Comment by Keith-CY** [comment]
> > > The `built-in` schema could be parsed and the first item could be active by default
> > > image
> > 
> > 
> > Any update on this suggestion?
> 
> Just saw that, it be implemented at these lines
> 
> [#339 (files)](https://github.com/Magickbase/ckb-explorer-frontend/pull/339/files#diff-51e3a32fe74b2924baa18f7966fb3867256d844a53c58339ffb037301f220b28R17-R20)

Got it, I will update it today

**Pull request merged**